tensorflow/python/data/experimental/kernel_tests/bucket_by_sequence_length_test.py

```diff
@@ -71,6 +71,130 @@ def _get_record_shape(sparse):
+      if sparse:
+        dataset = dataset.map(lambda x: (_to_sparse_tensor(x),))
+      return dataset
+
+    def _test_bucket_by_padding(no_padding):
+      dataset = build_dataset(sparse=no_padding)
+      dataset = dataset.apply(
+          grouping.bucket_by_sequence_length(
+              _element_length_fn,
+              boundaries,
+              batch_sizes,
+              no_padding=no_padding,
+              drop_remainder=True))
+      batch, = dataset.make_one_shot_iterator().get_next()
+
+      with self.cached_session() as sess:
+        batches = []
+        for _ in range(n_expected_batches):
+          batches.append(self.evaluate(batch))
+        with self.assertRaises(errors.OutOfRangeError):
+          self.evaluate(batch)
+
+      generated_lengths = []
+
+      # <seq-length>: <total-sum>
+      generated_sums = dict()
+
+      # <seq-length>: [<batch_size>, ...]
+      generated_batch_sizes = dict()
+
+      for length, batch_size, bucket_elements in zip(lengths,
+                                                     batch_sizes,
+                                                     n_bucket_elements):
+        # Initialize the sum across all batches.
+        generated_sums[length] = 0
+        # Initialize the individual batch sizes.
+        generated_batch_sizes[length] = []
+
+      for batch in batches:
+        shape = batch.dense_shape if no_padding else batch.shape
+        length = shape[1]
+        generated_lengths.append(length)
+
+        batch_size = shape[0]
+        generated_batch_sizes[length].append(batch_size)
+
+        batch_sum = batch.values.sum() if no_padding else batch.sum()
+        generated_sums[length] += batch_sum
+
+      for l in lengths:
+        # Make sure the sum of the batch contents is correct for the individual sequence lengths.
+        self.assertEqual(generated_sums[l],
+                         expected_sums[l],
+                         'Tensor sums did not match! '
+                         'expected: {}, generated: {}'
+                         .format(expected_sums, generated_sums))
+
+        # Make sure the individual batch sizes are generated as expected.
+        self.assertEqual(sorted(generated_batch_sizes[l]),
+                         sorted(expected_batch_sizes[l]),
+                         'Batch-sizes did not match! '
+                         'expected: {}, generated: {}'
+                         .format(sorted(expected_batch_sizes[l]),
+                                 sorted(generated_batch_sizes[l])))
+
+      # Make sure the generated sequence lengths appear as often as expected.
+      self.assertEqual(sorted(generated_lengths), sorted(expected_lengths),
+                       'The generated sequence lengths did not match! '
+                       'expected: {}, generated: {}'
+                       .format(sorted(expected_lengths),
+                               sorted(generated_lengths)))
+
+    for no_padding in (True, False):
```

Comment:
use `@parameterized` for this (see https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/experimental/kernel_tests/map_and_batch_test.py for an example)